### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by golden8807**
1. Candidate's details added patch version 
2. Validator's details added patch version 
3.Remove the cmdcode from Receipt.Topics hasno dependencies

### CODE REVIEW

**Comment by platonci** [comment]
Can one of the admins verify this patch?

**Comment by jiangshangqifeng** [comment]
ok to test

File: x/plugin/staking_plugin.go [hunk 1]
```diff
@@ -2167,7 +2167,8 @@ func (sk *StakingPlugin) ProposalPassedNotify(blockHash common.Hash, blockNumber
 				"blockHash", blockHash.Hex(), "nodeId", nodeId.String(), "err", err)
 			return err
 		}
-
+		//Store full version
+		can.ProgramVersion = programVersion
 		if err := sk.db.SetCanBaseStore(blockHash, addr, can.CandidateBase); nil != err {
 			log.Error("Failed to ProposalPassedNotify: Store CandidateBase info is failed", "blockNumber", blockNumber,
 				"blockHash", blockHash.Hex(), "nodeId", nodeId.String(), "err", err)
```
Comment (GavinXu520) [inline x/plugin/staking_plugin.go]: Is it better to comment out the line `version: = xutil.CalcVersion (programVersion)` in the methods `DeclarePromoteNotify` and` ProposalPassedNotify`?

Comment (golden8807) [reply x/plugin/staking_plugin.go]: It's great to make the previously changed code less intrusive to the original logic. I have modified it according to your proposal

**Comment by jiangshangqifeng** [comment]
rerun tests

**Pull request merged**